framework/src/modules/base_asset.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+/* eslint-disable class-methods-use-this */
+
+import { Schema } from '@liskhq/lisk-codec';
+import { StateStore as ChainStateStore, BlockHeader } from '@liskhq/lisk-chain';
+
+// Limit the scope of state store to which module can access
+export type StateStore = Omit<
+	ChainStateStore,
+	'consensus' | 'finalize' | 'createSnapshot' | 'restoreSnapshot'
```

Comment:
Why `consensus` is omitted here? Its also a state store, what if any module want to access the consensus information during transaction asset apply?

Reply:
consensus is only used for the internal information (like validators, finality), which should not be changed outside of framework.
That's why it is omitted, account/chain state should have enough information to be used in module/asset

Reply:
We can go with it, but I believe if someone develop a custom module working on different consensus logic, then he may need this. Not sure what use case it could be, just a thought.